src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -0,0 +1,412 @@
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.PRIMARY_ADDRESS_COUNTRY_BLANK);
+            allRequiredAreNotBlank = false;
+        }
+        if (StringUtils.isBlank(stgNewVendorRequestDetailToProcess.getRemittanceAddressCountry())) {
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.REMITTANCE_ADDRESS_COUNTRY_BLANK);
+            allRequiredAreNotBlank = false;
+        }
+        if (StringUtils.isBlank(stgNewVendorRequestDetailToProcess.getBankAddressCountry())) {
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.BANK_ADDRESS_COUNTRY_BLANK);
+            allRequiredAreNotBlank = false;
+        }
+        return allRequiredAreNotBlank;
+    }
+    
+    private boolean allRequiredFipsCountriesContainData(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
+        boolean allRequiredAreNotBlank = true;
+        if (StringUtils.isBlank(stgNewVendorRequestDetailToProcess.getRequestingCompanyTaxCountry())) {
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.FIPS_TAX_COUNTRY_BLANK);
+            allRequiredAreNotBlank = false;
+        }
+        return allRequiredAreNotBlank;
+    }
+    
+    private boolean isoCountryCodeTranslatesToSingleFipsCountryCode(String isoCountryCode, List<String> errorMessages) {
+        if (getPaymentWorksIsoToFipsCountryMap().containsKey(isoCountryCode)) {
+            if((getPaymentWorksIsoToFipsCountryMap().get(isoCountryCode)).size() == 1) {
+                return true;
+            }
+            else {
+                errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.SINGLE_ISO_MAPS_TO_MULTIPLE_FIPS + isoCountryCode);
+                return false;
+            }
+        }
+        else {
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBusinessRuleFailureMessages.ISO_COUNTRY_NOT_FOUND + isoCountryCode);
+            return false;
+        }
+    }
+    
+    private boolean pmwNewVendorIdentifierDoesNotExistInKfsStagingTable(PaymentWorksVendor stgNewVendorRequestDetailToProcess, List<String> errorMessages) {
+        if (getPaymentWorksVendorDao().isExistingPaymentWorksVendor(stgNewVendorRequestDetailToProcess.getPmwVendorRequestId())) {
+            errorMessages.add(PaymentWorksConstants.PaymentWorksBatchReportMessages.DUPLICATE_NEW_VENDOR_REQUEST_MESSAGE);
+            return false;
+        } 
+        else {
+            return true;
+        }
+    }
+    
+    private PaymentWorksVendor pmwPendingNewVendorRequestInitialSaveToKfsStagingTable(PaymentWorksVendor pmwVendor) {
+        LOG.info("pmwPendingNewVendorRequestSavesToKfsStagingTableSuccessful: entered");
+        pmwVendor.setPmwRequestStatus(PaymentWorksConstants.PaymentWorksNewVendorRequestStatusType.PENDING.getText());
```

Comment:
the part where the pmwVendor fields are updated could probably be extracted in a separate method

Reply:
Setting status values moved into new method as requested.